### PR TITLE
[2.0.0] Deploy all the APIs from Operator to Adapter using a Rest server

### PR DESCRIPTION
**Pull request by Jayanie**
## Purpose

- This PR introduces deploying all the APIs from Operator to Adapter using a REST server when the Adapter is starting or restarting.

## Approach 

- First the list of APIs which are deployed in kubernetes cluster is obtained.
- The zip files of the APIs are created
- From those zipped files one single zip file is created.

- When the Adapter restarts it will send a http request to Rest server endpoint of the Operator 
Example curl request: 
```
curl -X get "https://172.17.0.3:30445/operator/2.0/apis" -H "Authorization:Basic YWRtaW46YWRtaW4=" -k --output output
```
- That created zip file will be sent in the http response.

### CODE REVIEW

File: api-operator/deploy/controller-artifacts/operator-pem-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
...SWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTEhDWkRhdlpuV1BiWno0cGRBbDVyOGMKcjBwbWdUSnBId3Q4a0dMWnVRN1FmR2Z1anRNKytMZ2U2bVA2R2l4UFJnWTRCaW5QZWFEV2VBS0pJMUE2Z0hydwpycHI1aDJQNkZ6RDBQdStTa2hSQWxydlJoUmtKQ0VYQnJYMTVRaStUbVpHNTZIR1NmblVwOTAvb29qU3QzZU1YCkpOVE96S2hRLzY1U0Fnb1dBMHMvN1FIZmNGYzZ3OUxqc3A2cjFRb2FVc2h4VjNzOFlhRjdFWnVnYzF6dHNCWE8KTHlPNFJkSDJHVXRwNDlNeHhtcDRZWjJ0eUpicVZnQzZ5N2UwOGpCOCsxYkF0U3Flc3VENklkNDg5ZStGSm1ITApUdldvMEhPNFBTZHE4TDJpdW45SUZtRTZqenVXOFlSVEl3V3hNR0V5THZkWGsrdmZnVzhTK2tOT2Z3SFVUWmtDCkF3RUFBYU5UTUZFd0hRWURWUjBPQkJZRUZEYmY3QTdIVjllU1VNS2lkUEdIdnlKVzZocnpNQjhHQTFVZEl3UVkKTUJhQUZEYmY3QTdIVjllU1VNS2lkUEdIdnlKVzZocnpNQThHQTFVZEV3RUIvd1FGTUFNQkFmOHdEUVlKS29aSQpodmNOQVFFTEJRQURnZ0VCQUF1VGxTaFdPeDdiTjFKaHBzNndSZ1U2TThjdm1lWitMTGtTMkF3Wjc1c0tOcFQwCnF4WGloM1VrcUFaQ09JRXM5d0NaS1JuUkp1QlNPSzU1aDR4RjRLTnFleFRGa2JneE9NbmNTK014ZzY5czJuSWcKcm44ZkVIcldmdHVhM3Zzbk1kTmZrM0FqQ1hmbFJKYjB4NUo5dkRJUTdxeFNWVGVIZTRnTGFhMnV6elV2Y1hxTQpIMkJQS25WTUdpM2VwWEtIUnhvanlZTHVUbWZlRkJYQW5iRkJNK3NCREdnMWs5cUtDM0I2amxPVkQzZlhQRXFCCk5sUFBralJHSTlwSTgyTnBFSWU5VFdkN080MHNkMk02ZE1oYjhleVR0b2huRXBaRFoxSWRDSTkwNHJWVVNFaXAKalUwTnFHOXpnRXJiTEY3TGhydGFJMllHQkJwNUxWbFJ4Q3RHZVJrPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0t
```
Comment (pubudu538) [inline api-operator/deploy/controller-artifacts/operator-pem-secret.yaml:24]: WIthout having 2 secrets shall we use the secret relevant to the following kubectl command?

kubectl create secret tls ${CERT_NAME} --key ${KEY_FILE} --cert ${CERT_FILE}

File: api-operator/deploy/controller-artifacts/operator.yaml [hunk 2]
```diff
@@ -32,12 +32,21 @@ spec:
...
+          ports:
+            - containerPort: 9445
           command:
             - api-operator
-          args:
-            # Zap log level: "info" or "debug"
-            - "--zap-level=info"
+#          args:
+#            # Zap log level: "info" or "debug"
+#            - "--zap-level=info"
```
Comment (pubudu538) [inline api-operator/deploy/controller-artifacts/operator.yaml:49]: Lets remove commented code.

**Review by pubudu538 — CHANGES_REQUESTED** [review]

File: api-operator/deploy/controller-artifacts/operator-pem-secret.yaml (hunk 1 above)
Comment (Jayanie) [reply api-operator/deploy/controller-artifacts/operator-pem-secret.yaml]: Fixed with 25bb451d45419839cbeaebd100ce6d4d66e74413

File: api-operator/deploy/controller-artifacts/operator.yaml (hunk 2 above)
Comment (Jayanie) [reply api-operator/deploy/controller-artifacts/operator.yaml]: Fixed with 25bb451d45419839cbeaebd100ce6d4d66e74413

**Review by pubudu538 — APPROVED** [review]

File: api-operator/pkg/controller/api/api_controller.go [hunk 3]
```diff
@@ -120,6 +120,15 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
...
 	ctx = requestInfo.NewContext(ctx)
 
+	apiList := &wso2v1alpha2.APIList{}
+	if err := requestInfo.Client.List(ctx, apiList, client.InNamespace(common.WatchNamespace)); err != nil {
+		// Error reading the object - requeue the request.
+		reqLogger.Error(err, "Error reading all APIs in the specified namespace", "namespace",
+			common.WatchNamespace)
+		return reconcile.Result{}, err
+	}
+
```
Comment (HeshanSudarshana) [inline api-operator/pkg/controller/api/api_controller.go:130]: Remove unnecessary new line

Comment (Jayanie) [reply api-operator/pkg/controller/api/api_controller.go:130]: This is fixed in the latest branch.

**Pull request merged**